pkg/graveler/graveler.go

```diff
@@ -1474,9 +1474,20 @@ func (g *Graveler) Delete(ctx context.Context, repository *RepositoryRecord, bra
 		})
 }
 
+func (g *Graveler) deleteConsumer(ctx context.Context, repository *RepositoryRecord, branch *Branch, cachedMetaRangeID *MetaRangeID, keysChan <-chan Key) error {
+	for key := range keysChan {
+		err := g.deleteUnsafe(ctx, repository, branch, key, cachedMetaRangeID)
+		if err != nil {
+			return err
+		}
```

Comment:
AFAICT this will _still_ end up not fulfilling the DeleteObjects contract: all errors returned need to be wrapped in a DeleteError for NewMapDeleteErrors (called from [controller](https://github.com/treeverse/lakeFS/blob/4992d91d15983ea189f3d6025e1211db2fe2ebfa/pkg/api/controller.go#L194) and from the [S3 gateway](https://github.com/treeverse/lakeFS/blob/4992d91d15983ea189f3d6025e1211db2fe2ebfa/pkg/gateway/operations/deleteobjects.go#L127)).  Otherwise the call fails instead of returning success and a list of failures.

Also we end up not even trying to delete everything, which violates the DeleteObjects contract in another way.

Note that we do not have much leeway here to change semantics of the call, because the S3 gateway uses it and we have to be "S3 \"compatible\"".